shell/packages/sandstorm-db/migrations.js

```diff
@@ -556,6 +556,45 @@ function extractLastUsedFromApiTokenOwner() {
   });
 }
 
+function setUpstreamTitles() {
+  // Initializes the `upstreamTitle` and `renamed` fields of `ApiToken.owner.user`.
+
+  const apiTokensRaw = ApiTokens.rawCollection();
+  const aggregateApiTokens = Meteor.wrapAsync(apiTokensRaw.aggregate, apiTokensRaw);
+
+  // First, construct a list of all *shared* grains. We will need to do a separate update()
+  // for each of these, so we'd like to skip those which have nothing to update.
+  const grainIds = aggregateApiTokens([
```

Comment:
Would this be simpler with [`distinct()`](https://mongodb.github.io/node-mongodb-native/api-generated/collection.html#distinct)?